### PR TITLE
chore(dependencies): update to spring-boot 2 and dependency management via kork-BOM

### PR DESCRIPTION
**Pull request by cfieber**
We've run this successfully in our test environment

### CODE REVIEW

File: kayenta-signalfx/kayenta-signalfx.gradle [hunk 1]
```diff
@@ -109,21 +109,16 @@ integrationTest {
...
-  compile "com.netflix.spinnaker.orca:orca-core:$orcaVersion"
-
-  compile group: 'com.signalfx.public', name: 'signalfx-java', version: '0.0.48'
-
-  testCompile group: 'com.tngtech.java', name: 'junit-dataprovider', version: '1.13.1'
+  testImplementation 'com.tngtech.java:junit-dataprovider:1.13.1'
+  testImplementation project(":kayenta-standalone-canary-analysis")
 
   // Integration Test dependencies
   integrationTestCompile sourceSets.main.output
```
Comment (fieldju) [inline kayenta-signalfx/kayenta-signalfx.gradle]: I had to add the following additional lines to run the integration tests

```groovy
  integrationTestAnnotationProcessor platform("com.netflix.spinnaker.kork:kork-bom:$korkVersion")
  integrationTestAnnotationProcessor "org.projectlombok:lombok"
  integrationTestCompileOnly "org.projectlombok:lombok"
```

File: gradlew [hunk 2]
```diff
@@ -1,5 +1,21 @@
 #!/usr/bin/env sh
 
+#
+# Copyright 2015 the original author or authors.
```
Comment (fieldju) [inline gradlew]: Nit: looks like an IDE injected this, and needs to be updated or removed.

File: gradlew.bat [hunk 3]
```diff
@@ -1,3 +1,19 @@
+@rem
+@rem Copyright 2015 the original author or authors.
```
Comment (fieldju) [inline gradlew.bat]: Nit: looks like an IDE injected this, and needs to be updated or removed.

**Comment by fieldju** [comment]
image

⬆️ Not sure if just me, but running this locally I can no longer access the swagger api docs.

File: kayenta-graphite/kayenta-graphite.gradle [hunk 4]
```diff
@@ -168,19 +168,15 @@ integrationTest {
...
-    compile spinnaker.dependency('lombok')
+    implementation "org.apache.commons:commons-io:1.3.2"
 
-    compile "org.apache.commons:commons-io:1.3.2"
-
-    compile "com.netflix.spinnaker.orca:orca-core:$orcaVersion"
+ //   testImplementation "com.fasterxml.jackson.core:jackson-databind"
 
     // Integration Test dependencies
     integrationTestCompile sourceSets.main.output
```
Comment (fieldju) [inline kayenta-graphite/kayenta-graphite.gradle]:
```groovy
  integrationTestAnnotationProcessor platform("com.netflix.spinnaker.kork:kork-bom:$korkVersion")
  integrationTestAnnotationProcessor "org.projectlombok:lombok"
  integrationTestCompileOnly "org.projectlombok:lombok"
```

Will probably need to be added here as well.

File: kayenta-core/kayenta-core.gradle [hunk 5]
```diff
@@ -1,40 +1,27 @@
...
-  compile spinnaker.dependency("korkSwagger")
-  compile spinnaker.dependency('lombok')
-  compile spinnaker.dependency('spectatorApi')
+  implementation "redis.clients:jedis"
+  implementation "com.netflix.spinnaker.kork:kork-core"
+  implementation "com.netflix.spinnaker.kork:kork-web"
+  implementation "com.netflix.spectator:spectator-api"
 
   // TODO update korkSwagger?
   // Force a newer version than what comes with korkSwagger, so that the api docs page isn't broken.
```
Comment (fieldju) [inline kayenta-core/kayenta-core.gradle]: Is this still true with the Kork dep changes?

https://github.com/spinnaker/kork/commit/6d1cb51d652dcc1be1435bfb9ba8d28b3addefcb#diff-2ac723eaa3292cedf317c0d9e863ac20

Looks like springfox was ripped out and replaced, I wonder if that has to do with the swagger page not loading?

File: gradlew (hunk 2 above)
Comment (cfieber) [reply gradlew]: this is gradle's script, not ours

File: kayenta-signalfx/kayenta-signalfx.gradle (hunk 1 above)
Comment (cfieber) [reply kayenta-signalfx/kayenta-signalfx.gradle]: oh thanks, I'll update that

File: gradlew (hunk 2 above)
Comment (cfieber) [reply gradlew]: (specifically this is autogenerated when you run `gradlew wrapper` to update to the latest gradle version, so I don't really want to deviate from what that produces)

**Pull request merged**